protocol/chainlib/chain_router.go

```diff
@@ -234,8 +263,8 @@ func newChainRouter(ctx context.Context, nConns uint, rpcProviderEndpoint lavase
 	_, apiCollection, hasSubscriptionInSpec := chainParser.GetParsingByTag(spectypes.FUNCTION_TAG_SUBSCRIBE)
 	// validating we have websocket support for subscription supported specs.
 	webSocketSupported := false
-	for key := range supportedMap {
-		if key.IsRequirementMet(WebSocketExtension) {
+	for _, requirement := range supportedMap {
+		if requirement.IsRequirementMet(WebSocketExtension) {
```

Comment:
can break if true

Reply:
Fixed in df21e04521b181134ddcb5c44f9b57da270b98db